app/javascript/components/track/IterationSummary.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState, useRef } from 'react'
+import { TestRunStatus } from '../student/Editor'
+import { fromNow } from '../../utils/time'
+import { SubmissionMethodIcon } from './iteration-summary/SubmissionMethodIcon'
+import { TestsStatusSummary } from './iteration-summary/TestsStatusSummary'
+import { AnalysisStatusSummary } from './iteration-summary/AnalysisStatusSummary'
+import { IterationsChannel } from '../../channels/iterationsChannel'
+
+export type Iteration = {
+  id: number
+  idx: number
+  submissionMethod: SubmissionMethod
+  createdAt: Date
+  testsStatus: TestRunStatus
+  representationStatus: RepresentationStatus
+  analysisStatus: AnalysisStatus
+}
+
+export enum SubmissionMethod {
```

Comment:
@iHiD I don't know what these all are. I'll leave this to you.

Reply:
This signifies whether someone submitted via the CLI or the in-browser editor. In time there might be other methods (such as VSCode or Atom). At the moment, API covers everything that's not CLI.